iceoryx_hoofs/posix/vocabulary/source/file_name.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iox/file_name.hpp"
+
+namespace iox
+{
+namespace details
+{
+bool file_name_does_contain_invalid_characters(const string<platform::IOX_MAX_FILENAME_LENGTH>& value) noexcept
+{
+    const auto valueSize = value.size();
+
+    for (uint64_t i{0}; i < valueSize; ++i)
+    {
+        // AXIVION Next Construct AutosarC++19_03-A3.9.1: Not used as an integer but as actual character
+        const char c{value[i]};
```

Comment:
Since `operator[]` does bounds checks which are not required here, what do you think about using `c_str` or maybe add a `unchecked_at` to the `string`? Even better would be a `string::for_each(iox::function<void(char)>)` but that is over the top for this PR.

Anyway, since this is nothing which is called in a hop path, I would also be fine to keep it as is.